packages/babel-preset-env/src/built-in-definitions.js

```diff
@@ -1,140 +1,259 @@
-    toJSON: ["es6.date.to-json"],
+    anchor: ["es.string.anchor"],
+    big: ["es.string.big"],
+    bind: ["es.function.bind"],
+    blink: ["es.string.blink"],
+    bold: ["es.string.bold"],
+    codePointAt: ["es.string.code-point-at"],
+    concat: ["es.array.concat"],
+    copyWithin: ["es.array.copy-within"],
+    endsWith: ["es.string.ends-with"],
+    entries: ArrayNatureIterators,
+    every: ["es.array.is-array"],
+    fill: ["es.array.fill"],
+    filter: ["es.array.filter"],
+    find: ["es.array.find"],
+    findIndex: ["es.array.find-index"],
+    fixed: ["es.string.fixed"],
+    flags: ["es.regexp.flags"],
+    fontcolor: ["es.string.fontcolor"],
+    fontsize: ["es.string.fontsize"],
+    forEach: ["es.array.for-each", "web.dom-collections.for-each"],
+    includes: ["es.string.includes", "es.array.includes"],
+    indexOf: ["es.array.index-of"],
+    italic: ["es.string.italics"],
+    keys: ArrayNatureIterators,
+    lastIndexOf: ["es.array.last-index-of"],
+    link: ["es.string.link"],
+    match: ["es.string.match"],
+    map: ["es.array.map"],
+    name: ["es.function.name"],
+    padStart: ["es.string.pad-start"],
+    padEnd: ["es.string.pad-end"],
+    reduce: ["es.array.reduce"],
+    reduceRight: ["es.array.reduce-right"],
+    repeat: ["es.string.repeat"],
+    replace: ["es.string.replace"],
+    search: ["es.string.search"],
+    slice: ["es.array.slice"],
+    small: ["es.string.small"],
+    some: ["es.array.some"],
+    sort: ["es.array.sort"],
+    splice: ["es.array.splice"],
+    split: ["es.string.split"],
+    startsWith: ["es.string.starts-with"],
+    strike: ["es.string.strike"],
+    sub: ["es.string.sub"],
+    sup: ["es.string.sup"],
+    toISOString: ["es.date.to-iso-string"],
+    toJSON: ["es.date.to-json"],
+    toString: [
+      "es.object.to-string",
+      "es.date.to-string",
+      "es.regexp.to-string",
+    ],
+    trim: ["es.string.trim"],
+    values: ArrayNatureIterators,
+    __defineGetter__: ["es.object.define-getter"],
+    __defineSetter__: ["es.object.define-setter"],
+    __lookupGetter__: ["es.object.lookup-getter"],
+    __lookupSetter__: ["es.object.lookup-setter"],
   },
 
   staticMethods: {
     Array: {
-      from: "es6.array.from",
-      isArray: "es6.array.is-array",
-      of: "es6.array.of",
+      from: ["es.array.from", "es.string.iterator"],
+      isArray: "es.array.is-array",
+      of: "es.array.of",
     },
 
     Date: {
-      now: "es6.date.now",
+      now: "es.date.now",
     },
 
     Object: {
-      assign: "es6.object.assign",
-      is: "es6.object.is",
-      getOwnPropertySymbols: "es6.object.get-own-property-symbols",
-      setPrototypeOf: "es6.object.set-prototype-of",
-      values: "es7.object.values",
-      entries: "es7.object.entries",
-      getOwnPropertyDescriptors: "es7.object.get-own-property-descriptors",
+      assign: "es.object.assign",
+      create: "es.object.create",
+      defineProperty: "es.object.define-property",
```

Comment:
I wonder; will, in core-js 3, it be possible to omit including the polyfill for this if your target does not include a browser that needs it?

Right now, in core-js 2, even if you don't include it via preset-env, it'll be overwhelmingly likely that whatever feature you do include the polyfill of will indirectly require the defineProperty polyfill.

It's true that it's safer that way, but it's unfortunate that you can't get significantly smaller code size even when you're specifically being unsafe by using this instead of just loading all of `core-js`.

Reply:
No, if any polyfill requires this logic - it will be added as an internal module. And it's about almost all fundamental ES5 features.

It's related [this issue](https://github.com/zloirock/core-js/issues/388). You talk about the second option from [this comment](https://github.com/zloirock/core-js/issues/388#issuecomment-377818011). It's not hard for implementation, but it will cause too many problems.

For example, `Array#fill` from the issue. It requires not just `Object.defineProperty` for adding this polyfill, but, for example, `Object.create` for adding `Array#[@@unscopables]` where should be the name of this method. `Object.create` depends on `Object.defineProperties`, `Object.defineProperties` depends on `Object.getOwnPropertyNames`, etc.

So for working `es.array.fill` in IE8 we should add all of those dependencies. It's critically hard to determinate this list for manual adding. It's not so critical for `babel-preset-env`, but the list of internal dependencies could be changed in `core-js`, for example, after fixing any engine bug and it could be a breaking change for `babel-preset-env`.

I wanted to propose this issue to discuss before `core-js@4`. And it's the main problem which can be some time solved by dropping IE8 support in `core-js`.